cmd/containerd-shim-runc-v2/task/service.go

```diff
@@ -124,31 +124,35 @@ type service struct {
 }
 
 type containerProcess struct {
-	Container *runc.Container
-	Process   process.Process
+	Container          *runc.Container
+	Process            process.Process
+	PendingExecCounter *int
```

Comment:
I can see why this is a pointer, because `containerProcess` is passed by value. I wonder though if it would be a cleaner solution to just update `running` to make containerProcess a pointer. The previous assumption that it will not mutate so can pass by value is no longer true.